test/main.cpp

```diff
@@ -0,0 +1,32 @@
+#include <gtest/gtest.h>
+
+#include <torch/cuda.h>
+
+#include <iostream>
+#include <string>
+
+std::string add_negative_flag(const std::string& flag) {
+  std::string filter = ::testing::GTEST_FLAG(filter);
+  if (filter.find('-') == std::string::npos) {
+    filter.push_back('-');
+  } else {
+    filter.push_back(':');
+  }
+  filter += flag;
+  return filter;
+}
+
+int main(int argc, char* argv[]) {
+  ::testing::InitGoogleTest(&argc, argv);
+  if (!torch::cuda::is_available()) {
```

Comment:
Can we just make this an error? I don't see any use case where we want to potentially run with no CUDA.